src/main/java/software/amazon/nio/spi/s3/S3ClientStore.java

```diff
@@ -44,71 +19,26 @@ public class S3ClientStore {
 
     private static final S3ClientStore instance = new S3ClientStore();
 
-    /**
-     * Default S3CrtAsyncClientBuilder
-     */
-    protected S3CrtAsyncClientBuilder asyncClientBuilder = S3AsyncClient.crtBuilder();
-
-    /**
-     * Default client using the "https://s3.us-east-1.amazonaws.com" endpoint
-     */
-    @SuppressWarnings("JavadocLinkAsPlainText")
-    public static final S3Client DEFAULT_CLIENT = S3Client.builder()
-            .endpointOverride(URI.create("https://s3.us-east-1.amazonaws.com"))
-            .region(Region.US_EAST_1)
-            .build();
-
-    /**
-     * Default asynchronous client using the "https://s3.us-east-1.amazonaws.com" endpoint
-     */
-    @SuppressWarnings("JavadocLinkAsPlainText")
-    public static final S3AsyncClient DEFAULT_ASYNC_CLIENT = S3AsyncClient.builder()
-            .endpointOverride(URI.create("https://s3.us-east-1.amazonaws.com"))
-            .region(Region.US_EAST_1)
-            .build();
 
     private final Map<String, S3Client> bucketToClientMap = Collections.synchronizedMap(new HashMap<>());
     private final Map<String, S3AsyncClient> bucketToAsyncClientMap = Collections.synchronizedMap(new HashMap<>());
 
-    private final EqualJitterBackoffStrategy backoffStrategy = EqualJitterBackoffStrategy.builder()
-            .baseDelay(Duration.ofMillis(200L))
-            .maxBackoffTime(Duration.ofSeconds(5L))
-            .build();
+    Logger logger = LoggerFactory.getLogger("S3ClientStore");
 
-    final RetryCondition retryCondition;
-
-    {
-        final Set<Integer> RETRYABLE_STATUS_CODES = Stream.of(
-                HttpStatusCode.INTERNAL_SERVER_ERROR,
-                HttpStatusCode.BAD_GATEWAY,
-                HttpStatusCode.SERVICE_UNAVAILABLE,
-                HttpStatusCode.GATEWAY_TIMEOUT
-        ).collect(Collectors.toSet());
-
-        final Set<Class<? extends Exception>> RETRYABLE_EXCEPTIONS = Stream.of(
-                RetryableException.class,
-                IOException.class,
-                ApiCallAttemptTimeoutException.class,
-                ApiCallTimeoutException.class).collect(Collectors.toSet());
-
-        retryCondition = OrRetryCondition.create(
-                RetryOnStatusCodeCondition.create(RETRYABLE_STATUS_CODES),
-                RetryOnExceptionsCondition.create(RETRYABLE_EXCEPTIONS),
-                RetryOnClockSkewCondition.create(),
-                RetryOnThrottlingCondition.create()
-        );
-    }
-
-    final Logger logger = LoggerFactory.getLogger("S3ClientStore");
+    private static final S3ClientProvider DEFAULT_PROVIDER = new S3ClientProvider();
+    public static final S3Client DEFAULT_CLIENT = DEFAULT_PROVIDER.universalClient(false);
+    public static final S3AsyncClient DEFAULT_ASYNC_CLIENT = DEFAULT_PROVIDER.universalClient(true);
 
     private S3ClientStore(){}
 
     /**
      * Get the ClientStore instance
      * @return a singleton
      */
+    @Deprecated
```

Comment:
Can you update the Java docs with the new pattern that developers would use instead of this deprecated method?